modules/Kconfig.simplelink

```diff
@@ -13,8 +13,8 @@ config SIMPLELINK_HOST_DRIVER
 	depends on MULTITHREADING
 	select NEWLIB_LIBC
 	select ERRNO
-        select POSIX_API
         select PTHREAD_IPC
+	select POSIX_CLOCK
```

Comment:
Thanks for grouping my commit with yours. You may want to update the commit message to include clock and sleep:

```
PR #18780 introduces a way to decouple pthread support from the general
CONFIG_POSIX_API global switch. This commit modifies the build of
SimpleLink components to take advantage of it, since SimpleLink
libraries only require pthread, sem, clock, and sleep support, not
entire POSIX.

This fixes the build errors in the http_get sample introduced
by the merge of #18736.

```

Reply:
Ok, used this message. We also should remove that "platform_exclude: cc3220sf_launchxl" from http_get now, and to not go too detailed, I squashed that into this commit too.

Reply:
Thanks @pfalcon . LGTM